src/algebra/lie/of_associative.lean

```diff
@@ -66,6 +66,18 @@ lemma of_associative_ring_bracket (x y : A) : ⁅x, y⁆ = x*y - y*x := rfl
 
 end lie_ring
 
+instance lie_ring_module.of_associative_module {M : Type w} [add_comm_group M] [module A M] :
+  lie_ring_module A M :=
+{ bracket     := (•),
+  add_lie     := add_smul,
+  lie_add     := smul_add,
+  leibniz_lie :=
+    by simp [lie_ring.of_associative_ring_bracket, sub_smul, mul_smul, sub_add_cancel], }
+
+lemma lie_ring_module.of_associative_module_apply {M : Type w} [add_comm_group M] [module A M]
```

Comment:
Maybe this lemma should just be `lie_eq_smul`, to match `smul_eq_mul` etc?

Reply:
... Which raises an ugly question - doesn't this mean that for `a b : A` we have `⁅a, b⁆ = a * b`, but via another instance `a * b - b * a`? This looks like a diamond we can't have to me.

Reply:
Urgh, well spotted; this is ugly. Given a ring `A`, we would have two mathematically different `has_bracket A A`s:

1. `@ring.has_bracket A _` which says `⁅a, b⁆ = a * b - b * a`
2. `(@lie_ring_module.of_associative_module A _ A _ _).to_has_bracket` which says `⁅a, b⁆ = a • b` (and thus ` = a * b`)

I suppose the solution is to introduce a type synonym for `M` though another (more radical) solution would be to introduce a new typeclass to be used in place of `has_bracket` for `lie_ring_module`s.

Having said this, let me just double check: is there any chance we could live with this diamond, especially if we set the priority of `lie_ring_module.of_associative_module` lower than `ring.has_bracket`?

Reply:
Priorities aren't really a solution to diamonds unfortunately. The easy way out is to just make this `instance` a `def`, stick a warning on it in the docstring, and locally make it an instance in the file where you want to do stuff with this instance.

Reply:
Thanks again for spotting this.

I've opted for your "easy way out" especially as the main use case is for `A = module.End R M` and I can (and now have) safely make the corresponding instances in this case.